src/store/authors/reducer.js

```diff
@@ -0,0 +1,39 @@
+import {
+	ADD_COURSE_AUTHOR,
+	CREATE_AUTHOR,
+	REMOVE_COURSE_AUTHOR,
+	RESET_COURSE_AUTHORS,
+	SAVE_AUTHORS,
+} from './actionTypes';
+
+const authorsInitialState = { authors: [], courseAuthors: [] };
+
+export default function authorsReducer(
+	state = authorsInitialState,
+	{ type, payload }
+) {
+	switch (type) {
+		case SAVE_AUTHORS: {
+			return { ...state, authors: payload };
+		}
+		case CREATE_AUTHOR: {
+			return { ...state, authors: [...state.authors, payload] };
+		}
+		case ADD_COURSE_AUTHOR: {
```

Comment:
Courses authors logic should present in component only and shouldn't put anything to the store. Please see the comment above about initial store structure. Courses authors shouldn't present here

Reply:
got it